lib/components/shopping-cart-widget-component.js

```diff
@@ -10,96 +10,41 @@ export default class ShoppingCartWidgetComponent extends BaseContainer {
 		super( driver, by.className( 'cart-toggle-button' ) );
 	}
 
-	open() {
-		let d = webdriver.promise.defer();
+	async open() {
 		const driver = this.driver;
-		const explicitWait = this.explicitWait;
-
-		this.displayedOpen().then(
-			function cartAlreadyOpen() {
-				d.fulfill( true );
-			},
-			function cartClosed() {
-				driverHelper
-					.clickWhenClickable( driver, by.css( '.cart-toggle-button' ), explicitWait )
-					.then(
-						function success() {
-							d.fulfill( true );
-						},
-						function failure() {
-							d.reject( 'Unable to find Shopping Cart Widget button' );
-						}
-					);
-			}
-		);
-
-		return d.promise;
+		const explicitWait = this.explicitWaitMS;
+
+		if ( ! await this.displayedOpen() ) {
+			return await driverHelper.clickWhenClickable(
+				driver,
+				by.css( '.cart-toggle-button' ),
+				explicitWait
+			);
+		}
 	}
 
-	displayedOpen() {
-		let d = webdriver.promise.defer();
-
-		driverHelper.isElementPresent( this.driver, by.css( 'cart-body' ) ).then(
-			function( present ) {
-				if ( present ) {
-					d.fulfill( true );
-				} else {
-					d.reject( 'Cart not open' );
-				}
-			},
-			function() {
-				d.reject( 'Cart not open' );
-			}
-		);
-
-		return d.promise;
+	async displayedOpen() {
+		return await driverHelper.isElementPresent( this.driver, by.css( 'cart-body' ) );
```

Comment:
It's great to have one line, but that selector doesn't look like it should have ever worked as it's not a CSS class selector and there's no cart-body element in the DOM. Would you mind updating this to a proper CSS class selector - you could even re-use the selector from the constructor if you wanted to

Reply:
I removed displayedOpen(), and in open() method only click to open shopping cart (since in this flow cart is always closed and there is no need to check is it open, I removed check is it open).

Reply:
Sorry, I'm not saying what you did is wrong, I just noticed that `by.css( 'cart-body' )` would never had worked I don't think 🤔

Reply:
I think it should be `by.css( '.cart-body' )`

Reply:
Oh, I misread it again, I think that makes sense as we shouldn't have to worry about whether it's open 😀

Reply:
Yes, I understood. But it crossed my mind this would be more optimal solution? It will work both ways.